spec/models/bot_spec.rb

```diff
@@ -17,4 +17,141 @@
 
   it { should have_valid(:age).when("34") }
   it { should_not have_valid(:age).when(nil, "", 12, 204) }
+
+  let!(:user) { FactoryGirl.create(:user) }
+  let!(:bot) { FactoryGirl.create(:bot, user_id: user.id) }
+
+  before(:each) do
+    Bot.create(
+      name: 'Louisa',
+      gender: "Female",
+      eye_color: "Green",
+      hair_color: "Blonde",
+      age: 26,
+      user: user
+    )
+
+    interaction1 = Interaction.create(
+      category: "Sentence",
+      sentence: "You are great",
+      response: "I have a great teacher",
+      user: user
+
+    )
+
+    interaction2 = Interaction.create(
+      category: "Sentence",
+      sentence: "I want a bike",
+      response: "I think you should get one",
+      user: user
+    )
+
+    interaction3 = Interaction.create(
+      category: "Keyword",
+      sentiment: "Positive",
+      keyword1: 'cars',
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.